api.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type RecurseAPI struct {
+	rcAPIURL string
+}
+
+type RecurserProfile struct {
+	Stints []Stint
+}
+
+type Stint struct {
+	In_progress bool
+}
+
+func (ra *RecurseAPI) userIsCurrentlyAtRC(accessToken string, email string) bool {
+	currentlyAtRC := false
+
+	resp, err := http.Get(ra.rcAPIURL + "/profiles/" + email + "?access_token=" + accessToken)
+
+	if err != nil {
+		log.Printf("Got the following error while checking if the user was active at RC: %s\n", err)
+		return currentlyAtRC
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	//Parse the json response from the API
+	profile := RecurserProfile{}
+	json.Unmarshal([]byte(body), &profile)
+
+	log.Println("Recurser Profile: ", profile)
+
+	currentlyAtRC = profile.Stints[0].In_progress
+
+	log.Println("in_progress:", currentlyAtRC)
+
+	return currentlyAtRC
+}
+
+func (ra *RecurseAPI) getCurrentlyActiveEmails(accessToken string) []string {
+	//TODO, batch the API call since the limit is 50 results per
+
+	resp, err := http.Get(ra.rcAPIURL + "/profiles?scope=current&limit=50&role=recurser&access_token=" + accessToken)
+	if err != nil {
+		log.Printf("Got the following error while getting the RC batches from the RC API: %s\n", err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	//Parse the json response from the API
+	var recursers []map[string]interface{}
```

Comment:
Hmm, I could probably use the `RecurserProfile` struct defined above to parse the JSON response.